#! /bin/sh /usr/share/dpatch/dpatch-run
## 90_channelscan-0.1.0-1.5.10.dpatch by yolgecen at vdrportal.de
## http://vdrportal.de/board/thread.php?threadid=69815
##
## Thomas Günther <tom@toms-cafe.de>:
##   - Added compatibility to VDR < 1.5.10
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Changes for VDR >= 1.5.10.

@DPATCH@
diff -ur channelscan-0.1.0/filter.c channelscan-0.1.0_vdr-1.5.10/filter.c
--- channelscan-0.1.0/filter.c  2005-09-20 19:21:25.000000000 +0200
+++ channelscan-0.1.0_vdr-1.5.10/filter.c       2007-10-15 23:43:08.000000000 +0200
@@ -372,6 +372,10 @@
         int Dpids[MAXDPIDS + 1] = { 0 };
         char ALangs[MAXAPIDS + 1][MAXLANGCODE2] = { "" };
         char DLangs[MAXDPIDS + 1][MAXLANGCODE2] = { "" };
+#if VDRVERSNUM >= 10510
+        int Spids[MAXDPIDS + 1] = { 0 };
+        char SLangs[MAXDPIDS + 1][MAXLANGCODE2] = { "" };
+#endif
         int Tpid = 0;
         int NumApids = 0;
         int NumDpids = 0;
@@ -445,7 +449,11 @@
                 delete d;
                 }
             }
+#if VDRVERSNUM >= 10510
+        Channel->SetPids(Vpid, Vpid ? Ppid : 0, Apids, ALangs, Dpids, DLangs, Spids, SLangs, Tpid);
+#else
         Channel->SetPids(Vpid, Vpid ? Ppid : 0, Apids, ALangs, Dpids, DLangs, Tpid);
+#endif
         Channel->SetCaIds(CaDescriptors->CaIds());
         Channel->SetCaDescriptors(CaDescriptorHandler.AddCaDescriptors(CaDescriptors));
         }
